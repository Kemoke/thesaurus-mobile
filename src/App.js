import React from 'react';
import DictionaryView from "./views/DictionaryView/DictionaryView";
import {
    createBottomTabNavigator,
    createAppContainer,
    NavigationScreenConfigProps
} from 'react-navigation';
import {Ionicons} from '@expo/vector-icons'
import AddWordView from "./views/AddWordView/AddWordView";
import {ThemeProvider, Theme} from "react-native-elements";
import {SynonymContext} from './SynonymContext'

const routes = [
    {
        title: "Dictionary",
        component: DictionaryView,
        icon: 'ios-book'
    },
    {
        title: "Add Word",
        component: AddWordView,
        icon: 'ios-add'
    }
];

const bottomTabNavigator = createBottomTabNavigator(routes.reduce((routeMap, route) => {
    routeMap[route.title] = route.component;
    return routeMap;
}, {}), {
    defaultNavigationOptions: ({navigation}: NavigationScreenConfigProps) => ({
        tabBarIcon: ({focused, horizontal, tintColor}) => {
            const {routeName} = navigation.state;
            const route = routes.find((route) => route.title === routeName);
            return <Ionicons name={route.icon} size={25} color={tintColor}/>
        }
    }),
    tabBarOptions: {
        activeTintColor: "#3CBD3F"
    }
});

const AppContainer = createAppContainer(bottomTabNavigator);

const appTheme: Theme = {
    colors: {
        primary: "#3CBD3F"
    }
};

export default class App extends React.Component {
    state = {
        words: {}
    };

    addSynonym = (word, synonym) => {
        const {words} = this.state;
        if (!words[word]) {
            words[word] = []
        }
        if (words[word].indexOf(synonym) === -1) {
            words[word].push(synonym);
        }

        if (!words[synonym]) {
            words[synonym] = []
        }
        if (words[synonym].indexOf(word) === -1) {
            words[synonym].push(word)
        }
        this.setState(words);
    };

    render() {
        return (
            <ThemeProvider theme={appTheme}>
                <SynonymContext.Provider value={{words: this.state.words, addSynonym: this.addSynonym}}>
                    <AppContainer/>
                </SynonymContext.Provider>
            </ThemeProvider>
        );
    }
}

