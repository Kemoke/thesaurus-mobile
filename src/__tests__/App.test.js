import React from 'react';
import {shallow} from "enzyme";
import App from "../App";

describe('App', function () {
    it('should render correctly', function () {
        const component = shallow(
            <App/>
        );
        expect(component).toMatchSnapshot();
    });
    it('should add new synonym properly', function () {
        const component = shallow(
            <App/>
        );
        component.instance().addSynonym("test", "test2");
        expect(component.state("words")).toEqual({
            "test": ["test2"],
            "test2": ["test"]
        })
    });
    it('should append synonym word existing in map', function () {
        const component = shallow(
            <App/>
        );
        component.setState({
            words: {
                "test": ["test2"],
                "test2": ["test3", "test"],
                "test3": ["test2"],
            }
        });
        component.instance().addSynonym("test", "test3");
        expect(component.state("words")).toEqual({
            "test": ["test2", "test3"],
            "test2": ["test3", "test"],
            "test3": ["test2", "test"],
        })
    });
});