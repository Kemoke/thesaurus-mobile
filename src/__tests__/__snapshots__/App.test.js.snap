// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "primary": "#3CBD3F",
      },
    }
  }
>
  <ContextProvider
    value={
      Object {
        "addSynonym": [Function],
        "words": Object {},
      }
    }
  >
    <NavigationContainer />
  </ContextProvider>
</ThemeProvider>
`;
